### ISSUE TITLE
GarbageCollection: Assertion error while working with ShaderTerrainMeshes and multi-threading

### ISSUE THREAD

**Issue by thetestgame** [panda3d/panda3d#489]
I enabled the multi-threading pipeline on a terrain system we have been working on and it worked fine up until it went to stash/load in another terrain cell. Consistently this assertion error occurs.

```
Assertion failed: get_ref_count() == 0 at line 128 of c:\buildslave\sdk-windows-amd64\build\panda\src\pgraph\renderState.cxx
Assertion failed: !is_destructing() at line 117 of c:\buildslave\sdk-windows-amd64\build\panda\src\pgraph\renderState.cxx
Traceback (most recent call last):
Assertion failed:   File "D:\Dev\Project-Stars\build\game\client\src\Debug\direct\showbase\ShowBase.py", line 1974, in __garbageCollectStates
_node_ref_count != -100 at line 78 of c:\buildslave\sdk-windows-amd64\build\built\include\nodeCachedReferenceCount.I
Assertion failed: _cache_ref_count != -100 at line 78 of c:\buildslave\sdk-windows-amd64\build\built\include\cachedTypedWritableReferenceCount.IAssertion failed: obj->_flag == (AtomicAdjust::Integer)DCF_deleted, file c:\buildslave\sdk-windows-amd64\build\dtool\src\dtoolbase\deletedBufferChain.cxx, line 53
```

**Comment by rdb** [comment]
Is it possible to create a reproduction case, perhaps by modifications to the shader-terrain sample?  I've not been able to reproduce this one.

**Comment by rdb** [comment]
I think the fix in #499 should address this issue, right?

**Comment by thetestgame** [comment]
This seems to be fixed now. However it is worth noting that there is zero difference in performance from the application spending all the time in waiting now because of what looks like issue #951
